src/hotspot/share/runtime/os.cpp

```diff
@@ -1528,6 +1528,56 @@ bool os::set_boot_path(char fileSep, char pathSep) {
   return false;
 }
 
+static char* _image_release_file_content = nullptr;
+
+void os::read_image_release_file() {
+  const char* home = Arguments::get_java_home();
+  int rfile_len = (int)strlen(home) + 10;
+
+  char* release_file = (char*) os::malloc(rfile_len, mtInternal);
+  if (release_file) {
+    os::snprintf(release_file, rfile_len, "%s/release", home);
```

Comment:
Instead of the manual malloc, just do this:

```
stringStream ss;
ss.print(""%s/release", home);
```

then you can use `ss.base()` for the assembled path.